tensorboard/plugins/core/core_plugin.py

```diff
@@ -109,17 +113,34 @@ def get_resource_apps(self):
                     content = zip_.read(path)
                     # Opt out of gzipping index.html
                     if path == "index.html":
-                        apps["/" + path] = functools.partial(
-                            self._serve_index, content
-                        )
+                        for app_path in APP_PATHS:
+                            path_with_slash = (
+                                app_path
+                                if app_path.endswith("/")
+                                else app_path + "/"
+                            )
+                            apps[path_with_slash + path] = functools.partial(
+                                self._serve_index, content, app_path
+                            )
                         continue
 
                     gzipped_asset_bytes = _gzip(content)
                     wsgi_app = functools.partial(
                         self._serve_asset, path, gzipped_asset_bytes
                     )
-                    apps["/" + path] = wsgi_app
-        apps["/"] = apps["/index.html"]
+                    for app_path in APP_PATHS:
+                        path_with_slash = (
+                            app_path
+                            if app_path.endswith("/")
+                            else app_path + "/"
+                        )
+                        apps[path_with_slash + path] = wsgi_app
+        for app_path in APP_PATHS:
+            if app_path.endswith("/"):
+                apps[app_path] = apps[app_path + "index.html"]
+            else:
+                apps[app_path] = apps[app_path + "/index.html"]
+                apps[app_path] = apps[app_path]
```

Comment:
This DEFINITELY needs a unit test

Reply:
Unit test here would be great. I don't understand what line 143 does.